CARDS-1715: As a user landing on the PROMs login page, I can choose whether I wish to authenticate as a Patient or a Healthcare Provider

IntegralProgrammer opened the pull request:
This PR implements [CARDS-1715](https://phenotips.atlassian.net/browse/CARDS-1715) by providing the `<PromsLandingPage>` component and attaching it to the `loginPageStart` UI ExtensionPoint. Therefore, when CARDS is first accessed, a landing page is presented asking the user if they are a _Patient_ or a _Healthcare Provider_. If the user selects _Patient_, they are directed to `/Proms`. If the user selects _Healthcare Provider_, they are shown the standard Cards4Proms login page.

To test:

1. Build this `CARDS-1715` branch with `mvn clean install`.
2. Start CARDS in _Cards4Proms_ mode with `./start_cards.sh -P cards4proms --dev`.
3. Visit `http://localhost:8080`.
4. You should be shown a screen asking if you are a _Patient_ or a _Healthcare Provider_.
5. Clicking _I am a Patient_ should redirect the browser to `/Proms`. At this point it is expected to see an _Invalid access_ message as [CARDS-1716](https://phenotips.atlassian.net/browse/CARDS-1716) has not yet been implemented.
6. Back at the landing page (`http://localhost:8080/`), clicking _I am a Healthcare Provider_ should reveal the normal _DATA PRO_ login page.

IntegralProgrammer committed:
CARDS-1715: As a user landing on the PROMs login page, I can choose whether I wish to authenticate as a Patient or a Healthcare Provider

Implement the <PromsLandingPageDialog> component and attach it to the loginPageStart UI ExtensionPoint

sashaandjic commented:
tested - works as expected
<img width="669" alt="image" src="https://user-images.githubusercontent.com/23243244/163235291-d00d700b-56aa-4cc2-9460-37c21a92065f.png">

I have some minor cosmetic suggestions:
- change "I AM A" to "I am a"
- or have "I am a", then PATIENT below, and below it HEATHCARSEPROVIDER button
![image](https://user-images.githubusercontent.com/23243244/163235195-0d48ccc6-15f2-4288-a2c3-29264254b37b.png)

marta- committed:
CARDS-1715: As a user landing on the PROMs login page, I can choose whether I wish to authenticate as a Patient or a Healthcare Provider

Styling adjustments

marta- committed:
CARDS-1715: As a user landing on the PROMs login page, I can choose whether I wish to authenticate as a Patient or a Healthcare Provider
Use the configured appInfo instead of hard-coding "DATA-PRO"

marta- committed:
CARDS-1715: As a user landing on the PROMs login page, I can choose whether I wish to authenticate as a Patient or a Healthcare Provider
Simpler name for the Proms landing page component

marta- commented:
Made some UI adjustments, @sashaandjic please retest.

There remains the issue that the default login page flashes on the screen briefly before the proms landing page is displayed, but I guess that can be solved in the next iteration.

marta- committed:
CARDS-1715: As a user landing on the PROMs login page, I can choose whether I wish to authenticate as a Patient or a Healthcare Provider
Bugfix: removed unnecessary scrollbar at the bottom of the Proms landing page

sashaandjic reviewed (approved):
Hate to complain. 
Would just two buttons `Patients` and `Health Care Providers` look better?
With toolltips "Sign in as a patient" and "Sign in as a heath care provider".
No "Sign is as a:" text.

![image](https://user-images.githubusercontent.com/23243244/164244305-43699227-4e83-4167-912d-9cfd4c4e4824.png)

sashaandjic commented:
Is this behaviour expected? Clicking on the Back button behaves differently for patients

https://user-images.githubusercontent.com/23243244/164247160-469c1d78-9789-4e05-abfa-b7d5db0861de.mp4

Do we need our own `BACK` button in case the wrong one was clicked on? 

.

marta- committed:
CARDS-1715: As a user landing on the PROMs login page, I can choose whether I wish to authenticate as a Patient or a Healthcare Provider
Addressing comments: some additional UI adjustments

sashaandjic commented:
I also noticed that in test mode I no longer get the proper logo. Here is the screenshot of an instance running in `lfs` and `test` mode:
![image](https://user-images.githubusercontent.com/23243244/164295731-a0a8402f-1577-4325-a55b-37a29f147723.png)

sashaandjic reviewed (approved):
tested

marta- committed:
CARDS-1715: As a user landing on the PROMs login page, I can choose whether I wish to authenticate as a Patient or a Healthcare Provider
Clicking on the back button after navigating from the landing page to the HCP login now goes back to the landing page

marta- commented:
> Is this behaviour expected? Clicking on the Back button behaves differently for patients:

Fixed the behavior of the back button on the HCP login page, ready for retest @sashaandjic .

sashaandjic reviewed (approved):
re-tested

marta- commented:
The latest push is for rebasing only, no visible changes. @sdumitriu I recommend squash and merge, I made too many tiny changes in this PR.

marta- committed:
CARDS-1715: As a user landing on the PROMs login page, I can choose whether I wish to authenticate as a Patient or a Healthcare Provider
Bugfix: removed the loading effect when clicking on the landing page buttons, since some browsers will display the landing page in that state when returning to it by clicking the browser back button

sashaandjic reviewed (approved):
tested
